adapters/mock-da/src/service.rs

```diff
@@ -218,61 +274,17 @@ impl DaService for MockDaService {
     }
 
     async fn send_transaction(&self, blob: &[u8]) -> Result<(), Self::Error> {
-        let mut blocks = self.blocks.write().await;
-
-        let (previous_block_hash, height) = match blocks.iter().last().map(|b| b.header().clone()) {
-            None => (MockHash::from([0; 32]), 0),
-            Some(block_header) => (block_header.hash(), block_header.height + 1),
-        };
-        let data_hash = hash_to_array(blob);
-        // Hash only from single blob
-        let block_hash = block_hash(height, data_hash, previous_block_hash.into());
-
-        let blob = MockBlob::new(blob.to_vec(), self.sequencer_da_address, data_hash);
-        let header = MockBlockHeader {
-            prev_hash: previous_block_hash,
-            hash: block_hash,
-            height,
-            time: Time::now(),
-        };
-        let block = MockBlock {
-            header,
-            validity_cond: Default::default(),
-            blobs: vec![blob],
-        };
-        blocks.push_back(block);
-
-        // Enough blocks to finalize block
-        if blocks.len() > self.blocks_to_finality as usize {
-            let oldest_available_height = blocks[0].header().height();
-            let last_finalized_height = self.last_finalized_height.load(Ordering::Acquire);
-            let last_finalized_index = last_finalized_height
-                .checked_sub(oldest_available_height)
-                .unwrap();
-            let next_index_to_finalize = blocks.len() - self.blocks_to_finality as usize - 1;
-
-            if last_finalized_index > 0 {
-                assert_eq!(next_index_to_finalize as u64, last_finalized_index + 1);
-            }
-
-            self.finalized_header_sender
-                .send(blocks[next_index_to_finalize].header().clone())
-                .unwrap();
-
-            let this_finalized_height = oldest_available_height + next_index_to_finalize as u64;
-            self.last_finalized_height
-                .store(this_finalized_height, Ordering::Release);
-        }
-
+        let _ = self.add_blob(blob, Default::default()).await;
```

Comment:
We should probably evaluate the `Result` of `add_blob` via `self.add_blob(blob, Default::default()).await?;`